keyboards/handwired/do43/do43_oled.c

```diff
@@ -0,0 +1,293 @@
+/* Copyright 2021 Rocco Jiang
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "do43_oled.h"
+
+/*
+ * OLED drawing helper functions.
+ * Adapted from https://github.com/joric/qmk/blob/master/drivers/qwiic/micro_oled.c
+ */
+void oled_draw_line(uint8_t x0, uint8_t y0, uint8_t x1, uint8_t y1, bool on) {
```

Comment:
It might be worth pulling these out and porting them to the main oled driver code.

Reply:
For clarification – do you mean the code in `drivers/oled/oled_driver.h` and `ssd1306_sh1106.c`?

Reply:
Sorry, yes, I do.